### PR TITLE
Fix batch_norm_cpu to check the sizes of tensors

### PR DESCRIPTION
**Pull request by dulinriley**
When using `torch.ops.aten._native_batch_norm_legit.default` directly, if you accidentally pass the wrong shape of input tensors for weight, bias, or the stats, you get an ASAN exception on reading out of bounds from an accessor.
Instead, it should have a digestible error message that avoids memory corruption bugs.

Add a `TORCH_CHECK` that all of the inputs have the right dimensionality. This will raise a RuntimeError to python
if the check is violated. This behavior matches that of `F.batch_norm` which also uses `TORCH_CHECK` on the dimensionality
of its input tensors.

Added these error inputs as a unit test

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

**Comment by jerryzh168** [comment]
could you link the corresponding python checks as well?

**Comment by dulinriley** [comment]
@jerryzh168 not sure what you mean by the python checks, but I believe this code does the check for `F.batch_norm`:
https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/Normalization.cpp#L512-L527
It basically does the same check (and uses `TORCH_CHECK` for the exception creation).

As for tests, I don't see a `test/aten` directory. Is there a place for a C++ GTest or a good place for a python unittest for this type of batch norm functionality?

**Comment by jerryzh168** [comment]
> @jerryzh168 not sure what you mean by the python checks, but I believe this code does the check for `F.batch_norm`: https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/Normalization.cpp#L512-L527 It basically does the same check (and uses `TORCH_CHECK` for the exception creation).

yeah this is something I'm talking about, please update the summary to include this so we have more context

> 
> As for tests, I don't see a `test/aten` directory. Is there a place for a C++ GTest or a good place for a python unittest for this type of batch norm functionality?

I see, I think there is probably not test right now since it's not a public api, I'm not sure if it's legitimate to call this op directly in models actually, so we need to double check that as well

**Comment by jerryzh168** [comment]
@ezyang could you review this or find the right PoC?

**Review by ezyang — APPROVED** [review]
If you want to be super awesome, add an error input for these cases to the OpInfo, but it is not required to land

**Comment by dulinriley** [comment]
@ezyang what is the `OpInfo`? Where can I add these error cases?

**Comment by ezyang** [comment]
grep for your operator name in torch/testing/_internal/common_methods_invocations.py

**Comment by dulinriley** [comment]
Updated with a test change, I'll run it internally because the Anaconda build for OSS is confusing to me
